### ISSUE TITLE
Tumbler crash: URLError: <urlopen error [Errno 111] Connection refused>

### ISSUE THREAD

**Issue by dreboli** [JoinMarket-Org/joinmarket#444]
Tumbler crash: URLError: <urlopen error [Errno 111] Connection refused>

Is this a crash, or it would have been ok to just let the script run after the error message?

Latest commit was 440357abf04b8bbecfefec74ac5dfc10d0ab3ed4

```
python tumbler.py --addrcount=2 --wait-time=120 --donateamount=1 --liquiditywait=120 --mixdepthsource=2 tumblerwallet addr1 addr2
```

The log file is missing the the error I saw on console so here's both:
Console output:
```
47,501 [Thread-5    ] [DEBUG]  that was 1 tx out of 13
47,508 [Thread-5    ] [DEBUG]  ran unconfirmfun
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 57165)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/home/pi/joinmarket_0.1.3_develop/joinmarket/blockchaininterface.py", line 392, in __init__
    self, request, client_address, base_server)
  File "/usr/lib/python2.7/SocketServer.py", line 655, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
    method()
  File "/home/pi/joinmarket_0.1.3_develop/joinmarket/blockchaininterface.py", line 449, in do_HEAD
    urllib2.urlopen(request)
  File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 1227, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
URLError: <urlopen error [Errno 111] Connection refused>
----------------------------------------
48,841 [MainThread  ] [DEBUG]  <<pubmsg nick=Damipiver message=!absorder 0 41265 41265 0 0!absorder 1 79431 1630846 500 501!absorder 2 1630846 23355851 500 501 ~
51,447 [MainThread  ] [DEBUG]  <<pubmsg nick=Pujiyotaq message=!relorder 0 2502499 27347991 1000 0.0004 ~
^C08:16,858 [MainThread  ] [DEBUG]  CRASHING, DUMPING EVERYTHING
...
```

log file:
```
47,501 [Thread-5    ] [DEBUG]  that was 1 tx out of 13
47,508 [Thread-5    ] [DEBUG]  ran unconfirmfun
48,841 [MainThread  ] [DEBUG]  <<pubmsg nick=Damipiver message=!absorder 0 41265 41265 0 0!absorder 1 79431 1630846 500 501!absorder 2 1630846 23355851 500 501 ~
51,447 [MainThread  ] [DEBUG]  <<pubmsg nick=Pujiyotaq message=!relorder 0 2502499 27347991 1000 0.0004 ~
08:16,858 [MainThread  ] [DEBUG]  CRASHING, DUMPING EVERYTHING
08:16,867 [MainThread  ] [DEBUG]  Class debug dump, name:Wallet
...
08:18,933 [MainThread  ] [DEBUG]  Traceback (most recent call last):
  File "tumbler.py", line 576, in main
    irc.run()
  File "/home/pi/joinmarket_0.1.3_develop/joinmarket/irc.py", line 632, in run
    line = self.fd.readline()
  File "/usr/lib/python2.7/socket.py", line 447, in readline
    data = self._sock.recv(self._rbufsize)
  File "/usr/lib/python2.7/ssl.py", line 714, in recv
    return self.read(buflen)
  File "/usr/lib/python2.7/ssl.py", line 608, in read
    v = self._sslobj.read(len or 1024)
KeyboardInterrupt
```

.bitcoin/debug.log 

```
:46 AddToWallet 85censored new
:47 runCommand error: system(curl -sI --connect-timeout 100 http://localhost:62602/walletnotify?85censored) returned 13312
```

**Comment by chris-belcher** [comment]
Likely caused by this https://github.com/JoinMarket-Org/joinmarket/commit/97f032298187c2f2222cb8e8bb536b80cd5ec69d

Will probably be fixed by making it tolerate refused connection.

It's ok, that's what the develop branch is for. Thanks for the report.

**Comment by adlai** [comment]
Fixed by #445
